### PR TITLE
Bootloader and upstream updates

### PR DESCRIPTION
**Pull request by themarpe**
This PR adds the following changes to XLink:
 - bootloader support (added new device, with a custom name: '[path]-bootloader')
 - upstream XLink changes from openvino (most of diff is changing direct prints to mvLog)
 - XLinkLog: support for Windows (setting mvLogDefaultLevel to MVLOG_LAST disables all messages on Windows as well)
 - Some const qualifier on public facing API
 - Added CI for Win, Mac and Linux for easier development

### CODE REVIEW

File: pc/PlatformDeviceControl.c [hunk 1]
```diff
@@ -157,27 +113,64 @@ int XLinkPlatformBootRemote(deviceDesc_t* deviceDesc, const char* binaryPath)
...
+#if (!defined(_WIN32) && !defined(_WIN64))
+        rc = pcie_boot_device(*(int*)pcieFd, firmware, length);
+#else
+        rc = pcie_boot_device(pcieFd, firmware, length);
+#endif
+        pcie_close(pcieFd); // Will not check result for now
+        return rc;
+    } else if (deviceDesc->protocol == X_LINK_USB_VSC) {
+
+        char subaddr[28+2];
```
Comment (szabi-luxonis) [inline pc/PlatformDeviceControl.c:155]: use a `const int` variable for size?

File: pc/protocols/usb_boot.h [hunk 2]
```diff
@@ -21,6 +19,7 @@ extern int usb_loglevel;
 #define DEFAULT_UNBOOTVID           0x03E7
 #define DEFAULT_UNBOOTPID_2485      0x2485
 #define DEFAULT_UNBOOTPID_2150      0x2150
+#define DEFAULT_BOOTLOADER_PID      0xf63c
```
Comment (szabi-luxonis) [inline pc/protocols/usb_boot.h:22]: We could use `0xB001`=BOOT as PID, it's nicer IMO.
It's [not taken](http://www.linux-usb.org/usb.ids), at least for 0x03E7 VID.

Comment (Luxonis-Brandon) [reply pc/protocols/usb_boot.h:22]: Clever.  :-). Makes it very clear.

File: pc/PlatformDeviceControl.c (hunk 1 above)
Comment (themarpe) [reply pc/PlatformDeviceControl.c:155]: This is an upstream change (same as before actually just the function moved around) - not exactly sure why 28. I'd leave as is for now

**Comment by themarpe** [comment]
Merged to `develop`

**Pull request closed without merging**